colour/colorimetry/spectrum.py

```diff
@@ -53,6 +56,88 @@
            'zeros_spd',
            'ones_spd']
 
+DEFAULT_WAVELENGTH_DECIMALS = 10
+"""
+Default wavelength precision decimals.
+
+DEFAULT_WAVELENGTH_DECIMALS : int
+"""
+
+
+class SpectralMapping(ArbitraryPrecisionMapping):
+    """
+    Defines the base mapping for spectral data.
+
+    It enables usage of floating point wavelengths as keys by rounding them at
+    a specfic decimals count.
+
+    Parameters
+    ----------
+    data : dict or SpectralMapping, optional
+        Spectral data in a *dict* or *SpectralMapping* as follows:
+        {wavelength :math:`\lambda_{i}`: value,
+        wavelength :math:`\lambda_{i+1}`,
+        ...,
+        wavelength :math:`\lambda_{i+n}`}
+    wavelength_decimals : int, optional
+        Decimals count the keys will be rounded at.
```

Comment:
It is a regular condition I put when doing type checking on properties in general, it gives a chance for the underlying object to be initialised. I could remove the entire check block from here as it is only an alias for `self.key_decimals` which is itself also checking for the value to be an `int`.